### ISSUE TITLE
Performance issue

### ISSUE THREAD

**Issue by GSpecDrum** [Sylius/SyliusThemeBundle#109]
I have performance issues with migrating SyliusThemeBundle from liip/theme-bundle.

vendor | version
------------ | -------------
Symfony/Symfony | 4.4.15
Sylius/SyliusThemeBundle | 2.2.0
sonata-project/SonataAdminBundle | 3.76

Our project has more than 3 themes and we are using SonataAdminBundle. It has many twig template calls to rendering small blocks.

As a result, `Sylius\Bundle\ThemeBundle\Twig\ThemeFilesystemLoader::locateTemplate` is called more than 200 times per request.

Inside this method `HierarchicalTemplateLocator` rotating through all themes and all themed locators for each twig render call. It tooks very long time for such frequent calls.

In total, requests fall in timeout.

I have profiler screenshots to show my case:

![image](https://user-images.githubusercontent.com/28734885/132311076-294399f0-bf1c-4e7d-9f62-c9d252c4ea4c.png)
2 themes is configured

![image](https://user-images.githubusercontent.com/28734885/132311412-9881d00d-0529-45c4-af37-766710f677ac.png)
7 themes configured

Based on the above, for now I can't using this bundle for my project. I don't understand why this bundle is recommended replacing for liip/theme-bundle.

**Comment by Roshyo** [comment]
Indeed it loops a lot, but still the performances showed here are terrible compared to the ones I get.
With 4 themes enabled, 4800 times on the `CompositeTemplateLocator` but the Sylius home page still loads in 750ms approx. And the render time was `0.1094` seconds. So clearly acceptable, even more since this whole thing will get cached once the production mode will be enabled (once on prod mode, it reaches `0.0390` seconds of rendering time)

Are you sure it is not rather related to your environment ?

In any case, do you have some solution in mind ?

**Comment by GSpecDrum** [comment]
Yes, it is related to environment, like I said. On dev environment it is terribly worse than on production.

About solution - I recommend look how it was done in liip/theme-bundle. There was caching mechanism to not call locateTemplate for each time when template is rendering.

**Comment by mario-fehr** [comment]
Probably related to #70
